src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -7167,6 +7167,13 @@ internal static Expression InvokeMethod(MethodBase mi, DynamicMetaObject target,
                     {
                         argExprs[i] = Expression.Default(parameterType);
                     }
+                    else if (!parameters[i].HasDefaultValue && parameterType != typeof(object) && argValue == Type.Missing)
+                    {
+                        // If the method contains just [Optional] without a default value set then we cannot use
+                        // Type.Missing as a placeholder. Instead we use the default value for that type. Only
+                        // exception to this rule is when the parameter type is object.
+                        argExprs[i] = Expression.Default(parameterType);
```

Comment:
Please reword the comment.
It seems your comment in the PR description is more clear about behavior with `object`
> When it comes to the compiled assembly the ParameterInfo.DefaultValue for an argument with only [Optional] is [System.Reflection.Missing](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.missing?view=net-8.0) but this value cannot be passed as an argument unless the type is object because it cannot be casted to the argument type. Instead using null or the default value for that type should be used.

Reply:
I've updated the comment, please let me know if there's any further improvements or clarifications it should have.

Reply:
Ah sorry, that last host I made this change on was set to format on save, will undo those.

Reply:
Done